src/nimble.nim

```diff
@@ -156,7 +156,8 @@ proc processDeps(pkginfo: PackageInfo, options: Options): seq[PackageInfo] =
           "dependencies for $1@$2" % [pkginfo.name, pkginfo.specialVersion],
           priority = HighPriority)
 
-  var pkgList = getInstalledPkgsMin(options.getPkgsDir(), options)
+  var pkgList {.global.}: PackageFullInfoList = @[]
+  once: pkgList = getInstalledPkgsMin(options.getPkgsDir(), options)
```

Comment:
I just realised that this actually fixes #678, can you explain how this fix works? If we only ever get the list of installed packages once, then the list will not have the package we have installed so I would expect this to **cause** #678 not fix it. What am I missing?

Reply:
If you look at the code [below](https://github.com/nim-lang/nimble/blob/7c2b9f612e6a42609f28f8f51de4ab6949ced4a0/src/nimble.nim#L189) you will see that the next installed package is being added to the list:
```nim
pkgList.add((pkg, readMetaData(pkg.getRealDir())))
```
Frankly I don't like very much the mutually recursive way in which dependencies install works.
```
install -> installFromDir ->processDeps -> install
```
I can propose another approach which is `processDeps` to return a set of all unique dependencies from which to be subtracted the set of already installed packages and to install the missing ones with a simple for loop. Here is a pseudo code:
```nim
proc install(packageList) =
  var toInstall = toSet(packageList)
  # add all dependencies to the set
  # it is still recursive, but not mutually recursive with other procedures
  toInstall += processDeps(packageList) 
  let installedPackages = toSet(getInstalledPackages())
  toInstall -= installedPackages
  for package in toInstall:
    doInstall(package)
```
But I don't implement it in this way, because it will be too big change and easily something can be broken.

Reply:
> I think we need a `memoize` pragma in the stdlib.

I think that a pragma which behaves like C and C++ `static` keyword also will be appropriate. Then this can be on a single line.